adot-testbed/app/src/test/java/software/amazon/adot/testbed/LogsTests.java

```diff
@@ -65,61 +73,167 @@ private GenericContainer<?> createAndStartCollector(String configFilePath, Strin
+        File tempFile = new File(logDirectory.toString(), "storageExtension.log");
+        Thread.sleep(5000);
+
+        FileWriter fileWriter = new FileWriter(tempFile);
+        try {
+            fileWriter.write("First Message, collector is running" + "\n");
+            fileWriter.write("Second Message, collector is running" + "\n");
+            fileWriter.write("Third Message,  collector is running" + "\n");
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing to File A.", e);
+        }
+        fileWriter.flush();
+
+        List<String> logFilePaths = new ArrayList<>();
+        String expectedLogPath = logDirectory.toString();
+        logFilePaths.add(expectedLogPath + "/storageExtension.log");
+
+        validateLogs(logStreamName , logFilePaths, false);
+
+        collector.stop();
+
+        Thread.sleep(5000);
+
+        // write to the file when collector is stopped
+        try {
+            fileWriter.write("First Message after collector is stopped" + "\n");
+            fileWriter.write("Second Message after the collector is stopped" + "\n");
+            fileWriter.write("Third Message after the collector is stopped" + "\n");
+            fileWriter.flush();
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing to File A.", e);
+        }
+        fileWriter.close();
+       //restarting the collector
+        collector.start();
+
+        validateLogs(logStreamName , logFilePaths, false);
+        collector.stop();
+    }
+
+    @Test
+    void testFileRotation() throws Exception {
+        String logStreamName = "fileRotation-logstream-" + uniqueID;
+        collector = createAndStartCollector("/configurations/config-fileRotation.yaml", logStreamName);
+
+        Thread.sleep(5000);
+
+        // Create and write data to File A
+        File tempFile = new File(logDirectory.toString(), "testlogA.log");
+        try (FileWriter fileWriter = new FileWriter(tempFile)) {
+            fileWriter.write("Message in File A" + "\n");
+            fileWriter.flush();
+            fileWriter.close();
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing to File A.", e);
+        }
+        Thread.sleep(5000);
+
+        File renameFile = new File(logDirectory.toString(), "testlogA-1234.log");
+        tempFile.renameTo(renameFile);
+
+        File tempFileB = new File(logDirectory.toString(), "testlogA.log");
+        try (FileWriter newfileWriter = new FileWriter(tempFileB)) {
+            newfileWriter.write("Message in renamed file - line 1" + "\n");
+            newfileWriter.write("Message in renamed file - line 2" + "\n");
+            newfileWriter.write("Message in renamed file - line 3" + "\n");
+            newfileWriter.flush();
+            newfileWriter.close();
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing to File B.", e);
+        }
+        Thread.sleep(5000);
```

Comment:
again, I don't see a reason for sleeping here

Reply:
This thread can be removed, i was playing very safe i think in my local testing.